### DISCUSSION TITLE
Where to build Custom Activities, In Studio or workflow server ? Elsa 3

### DISCUSSION THREAD

**Discussion by GaryJurgens** [elsa-workflows/elsa-core#5215] [Q&A]
Good day

I am trying to port over Custom activities from Elsa 2,  where do you define them, in the studio, as we what the user to drag and drop or in the workflow server. I cannot seem to figure this out from the Documentation.

**Comment by glime-ai** [comment]
Good day,

To port over custom activities from Elsa 2 and make them available for users to drag and drop in the workflow designer, you need to define them in your .NET project and then ensure they are picked up by the Elsa server. Here are the general steps you would follow:

1. **Create the Custom Activity**: Write your custom activity class in C# by inheriting from the appropriate base class, such as `Activity` or `CodeActivity`. You can define input and output properties as needed.

2. **Register the Activity**: Register your custom activity with the Elsa services configuration in your `Startup.cs` or `Program.cs` file, depending on the version of .NET you are using. This makes the activity available to the Elsa runtime.

3. **Make the Activity Available in the Designer**: For the activity to show up in the Elsa Designer, you may need to create a corresponding front-end component if you are using Elsa 2. This typically involves writing some JavaScript to define how the activity should be rendered in the designer.

4. **Deploy the Activity**: Once your custom activity is created and registered, deploy your application with the Elsa workflow server. The custom activity should now be available in the Elsa Designer for users to drag and drop into their workflows.

Since you mentioned that you are having trouble finding this information in the documentation, it's worth noting that the documentation for Elsa is a work in progress and may not cover all scenarios in detail. However, the sources provided in the context, especially the one about [Custom Activities](https://v3.elsaworkflows.io/docs/extensibility/custom-activities), should give you a good starting point.

If you are using Elsa 3, the process might be slightly different, and the new designer may have different mechanisms for integrating custom activities. If you are looking for specific guidance on Elsa 3, it might be worth checking the Elsa 3 documentation or reaching out to the community for more detailed instructions.

Please let me know if you need further assistance or clarification on any of the steps.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
